internal/observability/call.go

```diff
@@ -261,17 +261,20 @@ func (c call) emitStreamError(err error) {
 
 // logStreamEvent is a generic logging function useful for logging stream
 // events.
-func (c call) logStreamEvent(err error, succMsg, errMsg string, extraFields ...zap.Field) {
+func (c call) logStreamEvent(err error, success bool, succMsg, errMsg string, extraFields ...zap.Field) {
```

Comment:
I personally think that the best way to fix it is:

if err == io.EOF {
      err = nil
}

when the context assumes that EOF is not an error. In this case don't need to add "success" param and confusing logic.

Reply:
I’m preserving the `err` so that it shows up in the error field of the subsequent log message.